### DISCUSSION TITLE
Best way to handle brain atlas in python in EEG case

### DISCUSSION THREAD

**Discussion by lemkis** [IS-UMK/EEG_brain_atlas#1] [Announcements]
#Reason
I want to write a lib in py for source reconstruction given EEG signal. To do so I need regions of interest (ROI) for a brain (at least i think so :)) and some "mapping" onto these regions. In short, (I think) I need some brain atlas. As a result of this discussion I d love to have object like class brain_atlas {...} which could load save and do some stuff on the brain atlasses + function taking brain_atlass as an argument which could find forward model, sth like https://mne.tools/dev/auto_tutorials/inverse/50_beamformer_lcmv.html#the-forward-model  but more universal.
#Questions
1. universal brain atlas for all subjects or brain atlas per subject? 
2. Does every brain atlas contain ROIs?
3. Should this atlas be 3d or 2d (2d in the sense of manifold)? 
4. Where to get such an atlas? 
5. How these atlases represented (mathematically) ? In particular, what is exactly (mathematically - given mathematical description of atlas) q in forward model  y = Hq + n where y corresponds to the electrods, q to the sources and n is some noise?
6. Is file format holding atlases unified or there are many? 
7. Is there sth like universal atlas for EEG? How to load it into python? 
8. Best python frameworks? MNE + Freesurfer ? Maybe some references/repos/example code?

**Comment by WlodzislawD** [comment]
Review on solving the inverse problem in EEG source analysis
https://jneuroengrehab.biomedcentral.com/articles/10.1186/1743-0003-5-25

**Comment by micholeodon** [comment]
Hi, here are my thoughts on your questions.

> 1. Universal brain atlas for all subjects or brain atlas per subject?

Depends. Usually it is better to have separate atlas for each subject. I mean, different regions differ in size and shape for different subject. It is best if atlas accounts for that, however it complicates the group analysis.
In such situation, aligning to some normalized space is usually helpful - see figures here: https://www.fieldtriptoolbox.org/tutorial/sourcemodel/#subject-specific_grids_that_are_equivalent_across_subjects_in_normalized_space

In ToFFi Toolbox, the user has a choice between using individual atlas (https://github.com/micholeodon/ToFFi_Toolbox/tree/master/ToFFi_Toolbox-20211013/DATA_PREPARATION/PARCELLATIONS_PREP/DK_individual_atlases/output_precomputed)
or group-level atlases (https://github.com/micholeodon/ToFFi_Toolbox/tree/master/ToFFi_Toolbox-20211013/commonData)

> 2. Does every brain atlas contain ROIs?

No. There are atlases that have "fuzzy" regions. More precisely, they are probabilistic maps that... well... map given point in space to a vector of probabilities, one coordinate for each brain region.
See https://www.fieldtriptoolbox.org/faq/how_is_the_segmentation_defined/
and
https://en.wikibooks.org/wiki/SPM/Atlases  

> 3. Should this atlas be 3d or 2d (2d in the sense of manifold)?

Depending on your research goal, you can divide brain into 2d regions (this is called parcellation) or 3d regions (called segmentation). 2D regions are concering the surface of the brain. 3D regions are used when deep sources and subcortical structures must be analysed as well. You can have atlases of both kind so pay attention to which atlas offer which level of description, i.e. surface or volume.

> 4. Where to get such an atlas?

Usually you download precomputed atlases and warp it to match individual brains. It can be done automatically using some toolboxes like Fieldtrip, Freesurfer. Sometimes you transform individual spatial coordinates of your subject to some common space like MNI. You can start here: https://www.fieldtriptoolbox.org/example/sourcemodel_aligned2mni/

> 5. How these atlases represented (mathematically) ? In particular, what is exactly (mathematically - given mathematical description of atlas) q in forward model y = Hq + n where y corresponds to the electrods, q to the sources and n is some noise?

Atlas is nothing more than a map between a point in 3d space and text label being the name of a given brain region. Usually these points are represented as a dicrete cubes of a given size, e.g. 2mm x 2mm x 2mm.

Now, the name of the region depends on how we decide do divide the brain, i.e. what is a division principle. Most basic are anatomical criterion (see Desikan-Killiany, AAL, Destrieux atlases), and functional criterion (see Schaeffer, Power atlases).
Here you can find helpful overview  - https://www.lead-dbs.org/helpsupport/knowledge-base/atlasesresources/cortical-atlas-parcellations-mni-space/.

This video will give you a better grip on the topic - https://www.youtube.com/watch?app=desktop&v=9yC5BQo98hM

It is also important to familiarize yourself with the concept of Talairach space and MNI space to know how brains of different subjects can be analyzed together (under what assumptions). Rigid, affine, and non-linear space transformations are used to this end.

> 6. Is file format holding atlases unified or there are many?

There are many formats. One of the most common is .nii. Some of the atlases are stored in .mat files.
Check here: https://www.fieldtriptoolbox.org/template/atlas/
Fieldtrip is very popular tool for fMRI and EEG/MEG analysis. I highly recommend their tutorials to get the idea how those different elements in source analysis (including atlases) are usually combined - https://www.fieldtriptoolbox.org/example/sourcemodel_aligned2mni_atlas/
 

> 7. Is there sth like universal atlas for EEG? How to load it into python?

There is no such thing as universal atlas. The closest term to this is group atlas, or template atlas. They usualy are a result of mulitple scans of a single subject  or average segmentation / parcellation of multiple subjects. See here - https://mcin.ca/research/neuroimaging-methods/atlases/ . 

I am not very familar in using EEG alases with Python.

> 8. Best python frameworks? MNE + Freesurfer ? Maybe some references/repos/example code?

The only one I know a bit is MNE. Freesurfer is mostly used for (f)MRI preprocessing - denoising, aligning, segmentation, etc. Not so much for source analysis.

Hope this helps :)

**Comment by siddharthprakash1** [comment]
Hello everyone,

I'm Siddharth, a college student with a keen interest in medical imaging and neural networks. I've recently come across Mr. Satrajit Ghosh's impressive 3D Brain Parcellation project on GitHub and I'm eager to learn more about it. Could you please offer your guidance and expertise on how to further develop and understand this project? Specifically, I'm interested in learning about 3D CNNs, dilated convolution, and residual connections. Additionally, any recommended tutorials or courses to help me grasp these concepts better would be greatly appreciated. Thank you all for your support and assistance.

Best regards,
Siddharth
